elf-size-analyze.py

```diff
@@ -1058,10 +1105,16 @@ def filter_symbols(section_key):
         Section.print(sections)
 
     if args.rom:
-        print_tree('ROM', filter_symbols(lambda sec: sec and sec.occupies_rom()))
+        if args.json:
+            create_json('ROM', filter_symbols(lambda sec: sec and sec.occupies_rom()))
+        else:
+            print_tree('ROM', filter_symbols(lambda sec: sec and sec.occupies_rom()))
```

Comment:
To avoid code duplication use:
```python
print_func = create_json if args.json else print_tree

if args.rom:
    print_func('ROM', filter_symbols(lambda sec: sec and sec.occupies_rom()))

if args.ram:
    print_func('RAM', filter_symbols(lambda sec: sec and sec.occupies_ram()))

if args.use_sections:
    # (...)
    print_func(name, filter_symbols(lambda sec: sec and sec.num in nums))
```